modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-react-fragments/fragments/policies-total-premium-due/index.js

```diff
@@ -12,176 +12,341 @@
  * details.
  */
 
-import React, {useEffect, useState} from 'react';
+import {ClaySelect} from '@clayui/form';
+import React, {useEffect, useRef, useState} from 'react';
 
 import BarChart from '../../../common/components/bar-chart';
 import ClayIconProvider from '../../../common/context/ClayIconProvider';
 import {getPoliciesChartExpiringPolicies} from '../../../common/services/Policy';
 import {getProducts} from '../../../common/services/Products';
 import {
 	currentDateString,
-	nextMonthDate,
+	lastDateCurrentMonth,
+	lastDateNextThreeMonth,
 } from '../../../common/utils/dateFormatter';
+import useWindowDimensions from '../../../hooks/useWindowDimensions';
+
+const PERIOD = {
+	THIS_MONTH: '1',
+	THREE_MONTH: '2',
+};
 
 export default function () {
-	const [policies, setPolicies] = useState([]);
 	const [expirationPoliciesList, setExpirationPoliciesList] = useState({});
-	const firstDayOfTheMonth = '01';
-	const policyExpirationDays = 15;
+	const [selectedFilterDate, setSelectedFilterDate] = useState('1');
+	const [dataChart, setDataChart] = useState();
+	const [labelChart, setLabelChart] = useState();
+	const [isLoading, setIsLoading] = useState(false);
+	const [sumPolicies, setSumPolicies] = useState();
 
-	function handleProductListFormatter() {
-		getProducts().then((response) => {
-			const productList = response?.data?.items;
+	const labelRef = useRef();
+	const {width} = useWindowDimensions();
 
-			productList.map((product) => {
-				product.productValue = 0;
-			});
+	const maxProductNameLenght = 10;
+	const minProductNameLenght = 7;
 
-			const productListFormatted = {};
-			productList.forEach((product) => {
-				if (!productListFormatted[product?.name]) {
-					productListFormatted[product?.name] = 0;
-				}
+	const chartContainer = document.getElementById('total-premium-due-bar');
 
-				setExpirationPoliciesList({...productListFormatted});
-			});
-		});
+	let chartWidth = 0;
+	if (chartContainer) {
+		const styles = getComputedStyle(chartContainer);
+		chartWidth = Number(styles.width.replace('px', ''));
 	}
 
-	function handleExpirationPolicies() {
-		getPoliciesChartExpiringPolicies(
-			currentDateString[0],
-			currentDateString[1],
-			firstDayOfTheMonth,
-			nextMonthDate[0],
-			nextMonthDate[1],
-			nextMonthDate[2]
-		).then((results) => {
-			const totalFilteredPolicies = results?.data;
+	const options = [
```

Comment:
Can you better specify the name of this **options**? For example: filterOptions.

Reply:
Done.